clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/ops/UpsertAppengineAutoscalingPolicyAtomicOperation.groovy

```diff
@@ -0,0 +1,96 @@
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.appengine.deploy.ops
+
+import com.google.api.services.appengine.v1.model.AutomaticScaling
+import com.google.api.services.appengine.v1.model.Operation
+import com.google.api.services.appengine.v1.model.Version
+import com.netflix.spinnaker.clouddriver.appengine.deploy.AppengineSafeRetry
+import com.netflix.spinnaker.clouddriver.appengine.deploy.description.UpsertAppengineAutoscalingPolicyDescription
+import com.netflix.spinnaker.clouddriver.appengine.deploy.exception.AppengineResourceNotFoundException
+import com.netflix.spinnaker.clouddriver.appengine.provider.view.AppengineClusterProvider
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import org.springframework.beans.factory.annotation.Autowired
+
+class UpsertAppengineAutoscalingPolicyAtomicOperation extends AppengineAtomicOperation<Void> {
+ private static final String BASE_PHASE = "UPSERT_SCALING_POLICY"
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  private final UpsertAppengineAutoscalingPolicyDescription description
+
+  @Autowired
+  AppengineClusterProvider appengineClusterProvider
+
+  @Autowired
+  AppengineSafeRetry safeRetry
+
+  UpsertAppengineAutoscalingPolicyAtomicOperation(UpsertAppengineAutoscalingPolicyDescription description) {
+    this.description = description
+  }
+
+  @Override
+  Void operate(List priorOutputs) {
+    task.updateStatus BASE_PHASE, "Initializing upsert of autoscaling policy for server group " +
+      "$description.serverGroupName in $description.credentials.region..."
+
+    def credentials = description.credentials
+    def serverGroupName = description.serverGroupName
+    def projectName = credentials.project
+
+    task.updateStatus BASE_PHASE, "Looking up $description.serverGroupName..."
+    def serverGroup = appengineClusterProvider.getServerGroup(credentials.name,
+                                                              credentials.region,
+                                                              serverGroupName)
+    def loadBalancerName = serverGroup?.loadBalancers?.first()
+
+    if (!serverGroup) {
+      throw new AppengineResourceNotFoundException("Unable to locate server group $serverGroupName.")
+    }
+    if (!loadBalancerName) {
+      throw new AppengineResourceNotFoundException("Unable to locate load balancer for $serverGroupName.")
+    }
+
+    def updatedAutoscalingPolicy = new AutomaticScaling(
+      minIdleInstances: description.minIdleInstances ?: serverGroup.scalingPolicy?.minIdleInstances,
+      maxIdleInstances: description.maxIdleInstances ?: serverGroup.scalingPolicy?.maxIdleInstances)
+    def version = new Version(automaticScaling: updatedAutoscalingPolicy)
+
+    task.updateStatus BASE_PHASE, "Setting min and max idle instance boundaries for $serverGroupName..."
+    safeRetry.doRetry(
```

Comment:
I wish. I'll make one. It's shared between GCE and App Engine, so it will be a lot of changes.